Compiling errors using ROCM7.0.x and clang 20.0.x

Trying to compile picongpu (dev branch ) example with ROCM7.0 i got the following compiling errors ( in attachment in this post )
Any idea what can go wrong here?

[errors.txt](https://github.com/user-attachments/files/23746891/errors.txt)

still compilation problems
```
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/EnergyFields.x.cpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/defines.hpp:24:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../pmacc/types.hpp:33:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC.hpp:43:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC_hostclass.hpp:31:
/lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/allocator.hpp:143:70: warning: volatile-qualified parameter type 'volatile size_t' (aka 'volatile unsigned long') is deprecated [-Wdeprecated-volatile]
  143 |         alloc(AlpakaDevice& dev, AlpakaQueue& queue, volatile size_t size)
      |                                                                      ^
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/BinEnergyParticles.x.cpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/algorithms/Gamma.hpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/algorithms/Gamma.def:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/defines.hpp:24:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../pmacc/types.hpp:33:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC.hpp:43:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC_hostclass.hpp:31:
/lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/allocator.hpp:143:70: warning: volatile-qualified parameter type 'volatile size_t' (aka 'volatile unsigned long') is deprecated [-Wdeprecated-volatile]
  143 |         alloc(AlpakaDevice& dev, AlpakaQueue& queue, volatile size_t size)
      |                                                                      ^
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/CountParticles.x.cpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/defines.hpp:24:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../pmacc/types.hpp:33:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC.hpp:43:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/mallocMC_hostclass.hpp:31:
/lustre/rz/dbertini/hydra/pog/picongpu/thirdParty/mallocMC/include/mallocMC/allocator.hpp:143:70: warning: volatile-qualified parameter type 'volatile size_t' (aka 'volatile unsigned long') is deprecated [-Wdeprecated-volatile]
  143 |         alloc(AlpakaDevice& dev, AlpakaQueue& queue, volatile size_t size)
      |                                                                      ^
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/EnergyFields.x.cpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/defines.hpp:37:
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/param/mallocMC.param:42:39: error: no member named 'FlatterScatterAlloc' in namespace 'mallocMC::CreationPolicies'
   42 |         = mallocMC::CreationPolicies::FlatterScatterAlloc::DefaultHeapConfig<blockSize, pageSize, wastefactor>;
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/plugins/CountParticles.x.cpp:22:
In file included from /lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/defines.hpp:37:
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/param/mallocMC.param:42:39: error: no member named 'FlatterScatterAlloc' in namespace 'mallocMC::CreationPolicies'
   42 |         = mallocMC::CreationPolicies::FlatterScatterAlloc::DefaultHeapConfig<blockSize, pageSize, wastefactor>;
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/param/mallocMC.param:46:37: error: no member named 'FlatterScatter' in namespace 'mallocMC::CreationPolicies'
   46 |         mallocMC::CreationPolicies::FlatterScatter<DeviceHeapConfig>,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/lustre/rz/dbertini/hydra/pog/picongpu/include/picongpu/../picongpu/param/mallocMC.param:46:37: error: no member named 'FlatterScatter' in namespace 'mallocMC::CreationPolicies'
   46 |         mallocMC::CreationPolicies::FlatterScatter<DeviceHeapConfig>,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
```

--- a/include/mallocMC/creationPolicies/Scatter.hpp
+++ b/include/mallocMC/creationPolicies/Scatter.hpp
@@ -1293,8 +1293,7 @@ namespace mallocMC
                     auto const gid = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc).sum();
 
                     auto const nWorker = alpaka::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc).prod();
-                    unsigned const temp
-                        = heapPtr->template getAvailaibleSlotsDeviceFunction(acc, numBytes, gid, nWorker);
+                    unsigned const temp = heapPtr->getAvailaibleSlotsDeviceFunction(acc, numBytes, gid, nWorker);
                     if(temp)
                         alpaka::atomicOp<alpaka::AtomicAdd>(acc, slots, temp);
                 };

